scanner/scanner.go

```diff
@@ -0,0 +1,150 @@
+package scanner
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"github.com/SkynetLabs/malware-scanner/clamav"
+	"github.com/SkynetLabs/malware-scanner/database"
+	"github.com/SkynetLabs/skynet-accounts/build"
+	"github.com/sirupsen/logrus"
+	"gitlab.com/NebulousLabs/errors"
+)
+
+var (
+	// sleepBetweenScans defines how long the scanner should sleep after
+	// scanning the DB and not finding any skylinks to scan.
+	sleepBetweenScans = build.Select(
+		build.Var{
+			Dev:      30 * time.Second,
+			Testing:  100 * time.Millisecond,
+			Standard: 10 * time.Second,
+		},
+	).(time.Duration)
+)
+
+// Scanner provides a convenient interface for working with ClamAV
+type Scanner struct {
+	staticCtx    context.Context
+	staticDB     *database.DB
+	staticClam   *clamav.ClamAV
+	staticLogger *logrus.Logger
+}
+
+// New returns a new Scanner with the given parameters.
+func New(ctx context.Context, db *database.DB, clam *clamav.ClamAV, logger *logrus.Logger) *Scanner {
+	return &Scanner{
```

Comment:
Should we have input validation here for `nil` inputs? is there a risk of panics?

Reply:
Checks added.
I don't think there was a real risk of panics with the current code but it doesn't hurt to be careful.